packages/ui/components/proposal/ProposalYourVote.tsx

```diff
@@ -1,5 +1,27 @@
-export interface ProposalYourVoteProps {}
+import clsx from 'clsx'
 
-export const ProposalYourVote = ({}: ProposalYourVoteProps) => {
-  return <p>placeholder</p>
+export type YourVoteProps = {
+  variant: 'abstain' | 'pending' | 'no' | 'yes'
+}
+
+export const ProposalYourVote = ({ variant }: YourVoteProps) => {
+  return (
+    <div
+      className={clsx(
+        'flex relative justify-center items-center py-1 px-4 w-20 text-sm font-medium rounded-full',
+        variant === 'abstain'
+          ? 'text-text-secondary bg-primary'
+          : variant === 'pending'
+          ? 'border-2 border-inactive'
+          : variant === 'no'
+          ? 'text-text-interactive-error bg-[#C73E5914]'
+          : 'text-text-valid bg-[#39A69914]'
+      )}
+    >
+      {variant[0].toUpperCase() + variant.slice(1)}
+      {variant === 'pending' && (
+        <div className="absolute top-0 right-0 w-2 h-2 bg-[#F53E86] rounded-full"></div>
```

Comment:
this hardcoding is necessary because the color (`#F53E86` = `rgb(245, 62, 134)`) does not exist in `index.css`